cli/azd/pkg/commands/builder.go

```diff
@@ -92,3 +112,11 @@ func RegisterDependenciesInCtx(
 
 	return ctx, nil
 }
+
+var _ azcore.TokenCredential = &panicCredential{}
+
+type panicCredential struct{}
+
+func (pc *panicCredential) GetToken(_ context.Context, _ policy.TokenRequestOptions) (azcore.AccessToken, error) {
+	panic("not logged in")
+}
```

Comment:
Do you see this `panicCredential` staying in the codebase or is it just for testing.  I question if it is needed if we have a proper control flow + custom errors, etc.

Reply:
This got reworked to make this a little clearer.  We should be able to remove all of this code once we stop trying to pull a credential out of the context object and instead have it injected into anything that needs it via wire.  Take another look and let me know if this makes sense.